File: native-app/android/app/src/main/java/com/callstack/nativehost/NativeShop.kt
package com.callstack.nativehost

import com.facebook.react.bridge.ReactApplicationContext
import com.rnshop.NativeShopSpec

class NativeShop(reactContext: ReactApplicationContext) : NativeShopSpec(reactContext) {

    companion object {
        const val NAME = "NativeShop"
    }

    override fun getItem(p0: String?): String {
        return "Hello from Native App!!!"
    }
}

File: src/specs/NativeShop.ts
import type {TurboModule} from 'react-native';
import {TurboModuleRegistry} from 'react-native';

export interface Spec extends TurboModule {
  getItem(key: string): string | null;
}

export default TurboModuleRegistry.getEnforcing<Spec>(
  'NativeShop',
);